draft-ietf-httpbis-message-signatures.md

```diff
@@ -382,7 +348,209 @@ This example shows a canonicalized value for the parameters of a given signature
   created=1618884475;expires=1618884775
 ~~~
 
-Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this field.
+Note that an HTTP message could contain multiple signatures, but only the signature parameters used for the current signature are included in this entry.
+
+### Method {#content-request-method}
+
+The `@method` identifier refers to the HTTP method of a request message. The value of is canonicalized by taking the value of the method as a string. Note that the method name is case-sensitive as per {{!SEMANTICS=RFC7231}}, and conventionally standardized method names are uppercase US-ASCII.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@method` value:
+
+~~~
+"@method": POST
+~~~
+
+If used in a response message, the `@method` identifier refers to the associated value of the request that triggered the response message being signed.
+
+### Target URI {#content-target-uri}
+
+The `@target-uri` identifier refers to the target URI of a request message. The value is the full absolute target URI of the request, potentially assembled from all available parts including the authority and request target.
+
+For example, the following message sent over HTTPS:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@target-uri` value:
+
+~~~
+"@target-uri": https://www.example.com/path?param=value
+~~~
+
+If used in a response message, the `@target-uri` identifier refers to the associated value of the request that triggered the response message being signed.
+
+### Authority {#content-request-authority}
+
+The `@authority` identifier refers to the authority component of the target of the HTTP request message. In HTTP 1.1, this is usually conveyed using the `Host` header, while in HTTP 2 and HTTP 3 it is conveyed using the `:authority` pseudo-header. The value is the fully-qualified authority component of the request, comprised of the host and, optionally, port of the request target, as a string.
+
+For example, the following request message:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@authority` value:
+
+~~~
+"@authority": www.example.com
+~~~
+
+If used in a response message, the `@authority` identifier refers to the associated value of the request that triggered the response message being signed.
+
+### Scheme {#content-request-scheme}
+
+The `@scheme` identifier refers to the scheme of the target URL of the HTTP request message. The value is the scheme as a string.
+
+For example, the following request message requested over plain HTTP:
+
+~~~
+POST /path?param=value HTTP/1.1
+Host: www.example.com
+~~~
+
+Would result in the following `@scheme` value:
+
+~~~
+"@scheme": http
```

Comment:
See above regarding c14n.  You need to choose a case here because some versions of HTTP don't carry a value.

Reply:
If there are good canonicalization rules we can point to from another document, that would be best to do -- we don't want to invent a new process here if we don't need to.

Reply:
These instructions have been added, pointing to the canonicalization rules in SEMANTICS specifically, which reference 3986 in turn.